reads_filtration/module_wraper.py

```diff
@@ -9,82 +11,173 @@
 from auxiliaries.stop_machine_aws import stop_machines
-from auxiliaries.validation_files import is_input_files_valid 
+from global_params import src_dir
+
+
+map_names_command_line = {
+    "fastq_path" : "fastq",
+    "parsed_fastq_results" : "reads_path",
+    "logs_dir" : "logs_dir",
+    "barcode2samplename" : "barcode2sample",
+    "done_file_path" : "done_path",
+    "left_construct" : "left_construct",
+    "right_construct" : "right_construct",
+    "max_mismatches_allowed" : "max_mismatches_allowed",
+    "min_sequencing_quality" : "min_sequencing_quality",
+    "minimal_length_required" : "minimal_length_required",
+    "multi_exp_config_reads" : "multi_experiments_config",
+    "check_files_valid" : "check_files_valid",
+    "stop_machines" : "stop_machines",
+    "type_machines_to_stop" : "type_machines_to_stop",
+    "name_machines_to_stop" : "name_machines_to_stop",
+    "gz" : "gz",
+    "verbose" : "v",
+    "error_path" : "error_path",
+    "queue" : "queue",
+    "rpm" : "rpm",
+    "mapitope" : "m"
+}
+
+
+def call_run_first_phase(dict_params, exp_name, argv):
+    run_first_phase(dict_params['fastq'], dict_params['reads_path'], dict_params['logs_dir'], dict_params['barcode2sample'], dict_params['done_path'],
+                    dict_params['left_construct'], dict_params['right_construct'], dict_params['max_mismatches_allowed'], dict_params['min_sequencing_quality'], 
+                    dict_params['minimal_length_required'],dict_params['check_files_valid'], dict_params['stop_machines'], dict_params['type_machines_to_stop'],
+                    dict_params['name_machines_to_stop'],dict_params['rpm'], dict_params['gz'], dict_params['v'],
+                    dict_params['m'], dict_params['error_path'], dict_params['queue'], exp_name, argv)
+
+
+def process_params(args, multi_experiments_config, argv):
+    # create data structure for running filter_reads
+    done_file = args.done_file_path
+    base_map =  args.__dict__
+    keys = base_map.keys()
+    base_map = change_key_name(base_map, map_names_command_line)
+    if multi_experiments_config:    
+        f = open(multi_experiments_config)
+        multi_experiments_dict = json.load(f)
+        # validation of the json file
+        is_valid = is_valid_json_structure(multi_experiments_config, multi_experiments_dict, schema_reads, logger)
+        if not is_valid:
+            return 
+        configuration = multi_experiments_dict['configuration']
+        base_map.update(configuration)
+        runs = multi_experiments_dict['runs']
+        for run in runs:
+            dict_params = base_map.copy()
+            dict_params.update(runs[run])
+            # create new list of argv of the specific run.
+            argv_new = []
+            argv_new.append(argv[0])
+            for k in keys:
+                val = str(dict_params[map_names_command_line[k]])
+                if (val != 'None') and (val != 'False'):
```

Comment:
I did not understand what I can convert to boolean.
I do str on all parameters because I can have numbers there too. And I want to put them all as a string to argv. And remove parameters that are false or None